discovery-provider/plugins/notifications/src/processNotifications/mappers/tastemaker.ts

```diff
@@ -0,0 +1,133 @@
+import { Knex } from 'knex'
+import { NotificationRow, PlaylistRow, TrackRow, UserRow } from '../../types/dn'
+import { TastemakerNotification } from '../../types/notifications'
+import { BaseNotification, Device, NotificationSettings } from './base'
+import { sendPushNotification } from '../../sns'
+import { ResourceIds, Resources } from '../../email/notifications/renderEmail'
+import { EntityType } from '../../email/notifications/types'
+
+type TastemakerNotificationRow = Omit<NotificationRow, 'data'> & {
+  data: TastemakerNotification
+}
+export class Tastemaker extends BaseNotification<TastemakerNotificationRow> {
+  receiverUserId: number
+  tastemakerItemId: number
+  tastemakerType: string
+  tastemakerUserId: number
+  tastemakerItemOwnerId: number
+
+  constructor(
+    dnDB: Knex,
+    identityDB: Knex,
+    notification: TastemakerNotificationRow
+  ) {
+    super(dnDB, identityDB, notification)
+    const userIds: number[] = this.notification.user_ids!
+    this.receiverUserId = userIds[0]
+    this.tastemakerItemId = this.notification.data.tastemaker_item_id
+    this.tastemakerItemOwnerId = this.notification.data.tastemaker_item_owner_id
+    this.tastemakerType = this.notification.data.tastemaker_item_type
+    this.tastemakerUserId = this.notification.data.tastemaker_user_id
+  }
+
+  async pushNotification() {
+    const res: Array<{
+      user_id: number
+      name: string
+      is_deactivated: boolean
+    }> = await this.dnDB
+      .select('user_id', 'name', 'is_deactivated')
+      .from<UserRow>('users')
+      .where('is_current', true)
+      .whereIn('user_id', [this.tastemakerUserId, this.receiverUserId, this.tastemakerItemOwnerId])
+    const users = res.reduce((acc, user) => {
+      acc[user.user_id] = {
+        name: user.name,
+        isDeactivated: user.is_deactivated
+      }
+      return acc
+    }, {} as Record<number, { name: string; isDeactivated: boolean }>)
+
+    if (users?.[this.receiverUserId]?.isDeactivated) {
+      return
+    }
+
+    const userNotifications = await super.getShouldSendNotification(
+      this.receiverUserId
+    )
+    const tastemakerUserName = users[this.tastemakerUserId]?.name
+    let entityType
+    let entityName
+
+    const track: { track_id: number; title: string } = await this.dnDB
+      .select('track_id', 'title')
+      .from<TrackRow>('tracks')
+      .where('is_current', true)
+      .whereIn('track_id', [this.tastemakerItemId]).first()
+
+
+    entityType = EntityType.Track
+    entityName = track.title
+
+
+    // If the user has devices to the notification to, proceed
+    if (
+      (userNotifications.mobile?.[this.receiverUserId]?.devices ?? []).length >
+      0
+    ) {
+      const devices: Device[] =
+        userNotifications.mobile?.[this.receiverUserId].devices
```

Comment:
might make this a const beforehand, then in if statement be like. `if (devices && devices.length > 0)`